shared/api/instance.go

```diff
@@ -28,6 +28,13 @@ type InstancesPost struct {
 	Type         InstanceType   `json:"type" yaml:"type"`
 }
 
+type InstancesPut struct {
```

Comment:
This needs a docstring comment above it or `golint` will fail.

Also, this needs another comment to indicate the API extension introducing it, look at `InstancesPost` below for an example.